testcases/commands/logrotate/logrotate_tests.sh

```diff
@@ -40,342 +40,148 @@
-crontab $LTPTMP/tst_logrotate.cron > $LTPTMP/tst_logrotate.out 2>&1 || RC=$?
-if [ $RC -ne 0 ]
-then
-    echo "Exit status of crontab command: $RC" >> tst_logrotate.out 2>/dev/null
-    tst_brk TBROK $LTPTMP/tst_logrotate.out NULL \
-        "Test #2: crontab Broke while installing cronjob. Reason:"
-    TFAILCNT=$(( $TFAILCN+1 ))
-else
-    tst_resm TINFO "Test #2: Cronjob installed successfully"
-fi
+test2(){
+        # Test #2
+        # Test that logrotate logrotate will rotate the logfile if the logfile
+        # exceeds a certain size.
+        #     - create a config file that will rotate the /var/log/tst_largelogfile.
+        #     - run logrotate in a cron job that runs every minute.
+        #     - add messages to the logfile until it gets rotated when a re-dittermined
+        #        size is reached.
+
+	cat >tst_largelog.conf <<-EOF
+        # create new (empty) log files after rotating old ones
+        create
+        # compress the log files
+        compress
+        # RPM packages drop log rotation information into this directory
+        include /etc/logrotate.d 
+        /var/log/tst_largelogfile {
+            rotate 5
+            size=2k
+        }
+	EOF
 
-# cron job to increase the log file size.
-cat >$LTPTMP/tst_addtolog.cron <<EOF
+	ROD chmod 644 tst_largelog.conf
 
-* * * * * echo "To Err Is Human, To Really Screw Up You Need A Computer."  >>/var/log/tst_largelogfile 2>/dev/null
-EOF
+	cat >/var/log/tst_largelogfile <<-EOF
+        # This is a psuedo-log file. This file will grow to a 2k size before
+        # getting rotated.
+	EOF
 
-tst_resm TINFO "Test #2: Installing cron job to increase logsize"
-crontab $LTPTMP/tst_addtolog.cron > $LTPTMP/tst_logrotate.out 2>&1 || RC=$?
-if [ $RC -ne 0 ]
-then
-    echo "Exit status of crontab command: $RC" >> tst_logrotate.out 2>/dev/null
-    tst_brk TBROK $LTPTMP/tst_logrotate.out NULL \
-        "Test #2: crontab Broke while installing cronjob. Reason:"
-    TFAILCNT=$(( $TFAILCN+1 ))
-else
-    tst_resm TINFO "Test #2: Cronjob installed successfully"
-fi
+	local logcontent="To Err Is Human, To Really Screw Up You Need A Computer."
+	for i in `seq 40`
+	do
+		echo "$logcontent" >> /var/log/tst_largelogfile
+	done
 
-# let cron jobs get started.
-sleep 10s
+	# cron job for logrotating
+	(crontab -l 2>/dev/null; echo \
+		"* * * * * ${LOGROTATE} $(pwd)/tst_largelog.conf") | crontab -
+	if [ $? -ne 0 ]; then
+		tst_brk TBROK "create cron job failed."
+	fi
 
-# increase the log file size.
+	# let cron jobs get started.
+	sleep 10s
 
-# wait for the /var/log/tst_largelogfile to be filled to a size greater than 2k
-tst_resm TINFO "Test #2: Checking if file size is > 2k"
-tst_resm TINFO "Test #2: Pls be patient this will take some time."
-tst_resm TINFO "Test #2: or killall -9 logrotate02 to skip.."
-if [ -f `which awk` ]
-then
-    while [ $filesize -lt 2046 ]
-    do
-        filesize=`ls -l /var/log/tst_largelogfile | awk '{print $5}'`
-    done
```

Comment:
Looping to check and sleep 1m is actually much better than sleep for 10m. You should use similar approach (cleaned). Also, `awk` will most likely be on the system which has `crontab` and `logrotate` installed.

Also, can we configure logrotate to run faster, so that it speedup the test?

Reply:
Hi pevik. So appreciate this get reviewed so quickly. 
1. I was thinking `stat` command is as common as `ls` and `awk` on linux. Should I change back to use `ls` and `awk`?
2. `sleep 10m` is already removed, the cleaned code generates logfile (larger than 2k) via cron job, so filesize check is appeard. I change to using just `echo` with a `for` loop to create a fixed size logfile. I'm not so much sure if this is good, any advise?
3. Yes, I'll try, maybe remove `sleep 10s`, and replace `sleep 1m` with `for` check.

Reply:
> Looping to check and sleep 1m is actually much better than sleep for 10m. You should use similar approach (cleaned). Also, `awk` will most likely be on the system which has `crontab` and `logrotate` installed.
> 
> Also, can we configure logrotate to run faster, so that it speedup the test?

Thanks for reply. I generate file size larger than 2k with `echo` and `for` loop, no more need to check file size. In the case do I must use a gradually increasing log file?